cms/static/js/views/course_video_settings.js

```diff
@@ -0,0 +1,545 @@
+                three_play_turnaround: self.selectedProvider === '3PlayMedia' ? self.selectedTurnaroundPlan : '',
+                preferred_languages: self.selectedLanguages,
+                global: false   // Do not trigger global AJAX error handler
+            }, function(data) {
+                if (data.transcript_preferences) {
+                    HtmlUtils.setHtml(
+                        $successEl,
+                        HtmlUtils.interpolateHtml(
+                            HtmlUtils.HTML('<div class="course-video-settings-success"><span class="icon fa fa-check-circle" aria-hidden="true"></span><span>{text}</span></div>'),
+                            {
+                                text: gettext('Settings updated')
+                            }
+                        )
+                    );
+                    self.activeTranscriptionPlan = data.transcript_preferences;
+                    // Sync ActiveUploadListView with latest active plan.
+                    Backbone.trigger('coursevideosettings:syncActiveTranscriptPreferences', self.activeTranscriptionPlan);
+                }
+            }).fail(function(jqXHR, textStatus, errorThrown) {
+                var errorMessage;
+                if (jqXHR.responseText) {
+                    // Enclose inside try-catch so that if we get erroneous data, we could still show some error to user
+                    try {
+                        errorMessage = $.parseJSON(jqXHR.responseText).error;
+                    } catch (e) {}
+                    HtmlUtils.setHtml(
+                        $errorEl,
+                        HtmlUtils.interpolateHtml(
+                            HtmlUtils.HTML('<div class="course-video-settings-error"><span class="icon fa fa-info-circle" aria-hidden="true"></span><span>{text}</span></div>'),
+                            {
+                                text: errorMessage || gettext('Error saving data')
+                            }
+                        )
+                    );
+                }
+            });
+        },
+
+        updateCourseVideoSettings: function(event) {
+            var $successEl = this.$el.find('.course-video-settings-success-wrapper');
+            if(this.validateCourseVideoSettings()) {
+                this.saveTranscriptPreferences();
+            } else {
+                $successEl.empty();
+            }
+        },
+
+        render: function() {
+            var dateModified = this.activeTranscriptionPlan ? moment.utc(this.activeTranscriptionPlan['modified']).format('ll') : '';
+            HtmlUtils.setHtml(
+                this.$el,
+                this.template({
+                    dateModified: dateModified
+                })
+            );
+
+            this.populatePreferences();
+
+            this.registerCloseClickHandler();
+            this.setFixedCourseVideoSettingsPane();
+            return this;
+        },
+
+        setFixedCourseVideoSettingsPane: function() {
```

Comment:
Instead of setting css, we should try to add/remove classes only. This way rtl/ltr things van be taken care of in css.

Reply:
I agree, you are right, I tried that way but couldn't. I am afraid the requirements are such that we have to mix these things here, otherwise it could have been simpler but keeping requirements in mind, I think this the only way we can do.

Reply:
And that's why I have added comments explaining too